crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/cheated_syscalls.rs

```diff
@@ -245,6 +248,65 @@ pub fn get_block_hash_syscall(
     Ok(GetBlockHashResponse { block_hash })
 }
 
+#[allow(clippy::needless_pass_by_value)]
+pub fn storage_read(
+    request: StorageReadRequest,
+    _vm: &mut VirtualMachine,
+    syscall_handler: &mut SyscallHintProcessor<'_>,
+    cheatnet_state: &mut CheatnetState,
+    _remaining_gas: &mut u64,
+) -> SyscallResult<StorageReadResponse> {
+    let contract_address = syscall_handler.storage_address();
+    let test_address = TryFromHexStr::try_from_hex_str(TEST_ADDRESS)?;
+
+    // Save the original storage_address
+    let original_storage_address = syscall_handler.base.call.storage_address;
+
+    if contract_address == test_address {
+        let cheated_data = cheatnet_state.get_cheated_data(contract_address);
+        if let Some(new_address) = cheated_data.contract_address {
+            syscall_handler.base.call.storage_address = new_address;
+        }
+    }
+
+    let value = syscall_handler.base.storage_read(request.address)?;
```

Comment:
If this error bubbles, there might be cases where original address is not restored. Does this error result in a fail or can it be handled? If the latter then it's problematic.